spec/npm/asciidoctor-all.spec.js

```diff
@@ -21,5 +37,76 @@ describe('Node.js', function () {
       expect(doc.attributes.$fetch('docname')).toBe('test.adoc');
     });
   });
+
+  describe('Converting file', function() {
+    it('should be able to convert a file', function() {
+      var expectFilePath = __dirname + '/test.adoc.html';
+      removeFile(expectFilePath);
+      try {
+        var doc = Asciidoctor.$convert_file(__dirname + '/test.adoc', null);
```

Comment:
There are a lot of different permutations here. You need to study the logic in the link below to understand the behavior:

https://github.com/asciidoctor/asciidoctor/blob/v1.5.4/lib/asciidoctor.rb#L1421-L1438